pkg/rollup/resolution.go

```diff
@@ -0,0 +1,182 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package rollup
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+const (
+	originalSchema          = "prom_data"
+	upperLimit              = 5000 // Maximum samples allowed
+	assumedScrapeInterval   = time.Second * 30
+	refreshRollupResolution = time.Minute * 30
+)
+
+type rollupInfo struct {
+	schemaName string
+	resolution time.Duration
+}
+
+type QueryHelper struct {
+	conn                pgxconn.PgxConn
+	ctx                 context.Context
+	refreshMtx          sync.RWMutex
+	metricType          map[string]string // metric_name: metric_type
+	resolutionsASC      []rollupInfo      // {schemaName, resolution} in ascending order of resolution.
+	downsamplingEnabled bool
+}
+
+func NewQueryHelper(ctx context.Context, conn pgxconn.PgxConn) (*QueryHelper, error) {
```

Comment:
Nit: naming here seems a bit vague. It seems that the only functionality that it provides is returning the `schema` name of rollups based on time range. Maybe `RollupManager` would be more appropriate?

Reply:
I did think to name manager, but then I couldn't conclude as what does it manage. So I thought in the direction that it helps to performing querying on rollups, so named `QueryHelper`. Now, I have changed to `Decider` since it helps to decide which `Schema` to be used based on {min, max}. Please let me know if the confusion regarding naming persists.